### PR TITLE
Pinned Tabs

### LINKED ISSUE #226: [FEATURE REQUEST] Pinned notes
Will it be hard to implement this option? Maybe have it appear on the file menu, like the "Set writing target...", and the pinned notes (note?) to appear on top of folders no matter how you choose to sort them?

### LINKED ISSUE #3348: [Enhancement] Pin Tabs
### Description

There are tabs that are more prominent than others. Its not uncommon that I copy fragments form several text files into one file.

Ok here is are real word examples why this would be useful for me:

1.) I navigate through my zettelkasten follow tags or links to gather information. This new Information I need for creation. IMHO this is the great benefit of a zettelkasten: It helps me to recreate content from my zettelkasten archive. 
2.) I often cluster (https://k12.thoughtfullearning.com/blogpost/dr-ricos-magic-key-clustering) for generating new, fresh ideas. this task i do almost every day. (the clusters shares all the same tag). After several Weeks I read all clusters and copy the Idea that i still find fresh in  a new file and create something new (e.g. a peom, a character for a story)

This tool already helps me to be creative, to focus on that what matters (nobody cares about formatting, spellchecking and other distraction stuff in the early stages of creation). This enhancement would lift up zettlr to a new level. 

### Proposed Changes

Like in the Webbrowser: right click on the tab, in the context menu "Pin Tab" (or unpin, if its already pinned)

### Caveats

No. You don't need to use it if you dont like it.

### Do you Wish to Attempt Implementing this Yourself?

No

### Zettlr Version

Stable (most recent version)

### Your Platform

- [ ] Windows
- [ ] macOS
- [X] Linux

### Operating System Version

Manjaro Linux, i3, packages up to date.

### Additional Information

_No response_

### PR DESCRIPTION
**Pull request by nathanlesage**
## Description

This PR adds the ability to pin and unpin documents, which prevents them from being accidentally closed.

## Changes

* The `openFiles` config array is now no longer a list of paths, but objects that hold the path as well as additional properties (as of now: a `pinned` property [boolean])
* The documents provider keeps the configuration up-to-date and manages that
* The config provider initially migrates the old array
* Pinned tabs have no longer a close button, instead a "pin" icon; they must be unpinned before they can be closed
* Pins are respected also for "broad" actions such as "close all other tabs" or "close all tabs"

## Additional information

Tested on: macOS 12.3.1 Monterey

This PR affects the following issues:

* Closes #3348
* Closes #226

### CODE REVIEW

**Comment by exanrauzer** [comment]
Hi! Do you consider to lock position of pinned tabs to left side and to exclude them from being sroll?

**Comment by nathanlesage** [comment]
That could be a next step. This PR first of all changes the data structure files are represented with and supplies the basic requirements for pinned tabs. This PR needs to be merged so that I can then, in a next PR, introduce split- and multiple windows. Afterwards, we can definitely go back to improving the pinned tabs. Feel free to already open an issue, laying out your improvements so that we don't forget them!

**Review by kyaso — COMMENTED** [review]
The delta looks fine to me. Unfortunately, I only have my iPad and won't have access to a computer for some time, so I couldn't test anything.

File: source/win-main/tabs-context.ts [hunk 1]
```diff
@@ -35,6 +36,15 @@ export default function displayTabsContext (event: MouseEvent, fileObject: MDFil
       label: trans('menu.tab_close_all'),
       id: 'close-all',
       type: 'normal',
+      enabled: !doc.pinned
+    },
+    {
+      type: 'separator'
+    },
+    {
+      label: doc.pinned ? 'Unpin tab' : 'Pin tab',
```
[the remark this answers is not in the document]
Comment (nathanlesage) [reply source/win-main/tabs-context.ts:45]: Good point, I'll get right to it.

**Comment by nathanlesage** [comment]
Thanks for having a look! It's perfectly fine that you couldn't test, I've been doing this extensively, and all potential bugs can be solved during beta, since the changes definitely merit a new major release at this point (especially with the next two PRs, which will introduce split tabs and multiple windows, and remove the Quicklook feature)

**Pull request merged**